File: apps/editor/src/structs/reward.rs
use serde_derive::{Deserialize, Serialize};

#[derive(Serialize, Deserialize, Debug, PartialEq)]
#[repr(C)]
pub struct RewardItem {
    pub rate: u16,
    pub item: u16,
    pub quantity: u16
}

#[derive(Serialize, Deserialize, Debug, PartialEq)]
#[repr(C)]
pub struct RewardTableHeader {
    pub table_id: u8,
    pub unk_0: u8,
    pub unk_1: u16,
    pub table_offset: u32
}

#[derive(Serialize, Deserialize, Debug, PartialEq)]
#[repr(C)]
pub struct RewardTable {
    pub table_header: RewardTableHeader,
    pub items: Vec<RewardItem>
}

File: packages/ui/lib/utils/quest-file/reward.ts

export interface RewardItem {
    rate: number;
    item: number;
    quantity: number;
}


export interface RewardTableHeader {
    table_id: number;
    unk_0: number;
    unk_1: number;
    table_offset: number;
}


export interface RewardTable {
    table_header: RewardTableHeader,
    items: RewardItem[];
}